Source/WebCore/html/canvas/CanvasRenderingContext2DBase.h

```diff
@@ -247,11 +248,14 @@ class CanvasRenderingContext2DBase : public CanvasRenderingContext, public Canva
     using Direction = CanvasDirection;
     void setDirection(Direction);
 
-    class FontProxy final : public FontSelectorClient {
-    public:
-        FontProxy() = default;
+    class FontProxy final : public RefCounted<FontProxy>, public FontSelectorClient {
```

Comment:
Why are we making this ref-counted exactly? Which problem are we trying to solve? It may be fine but we should have a good reason to pay the extra cost.

Reply:
The primary reason we want `FontProxy` to be ref-counted is to avoid making   `CanvasRenderingContext2DBase::drawTextUnchecked()` reference to `state().fontProxy` point to free'd memory, at any point the `m_stateStack` array gets re-allocated.

Reply:
We need to fix an invalid borrow problem for the `FontProxy`:
```
auto& font = *this->fontProxy();
save();
font.metricsOfPrimaryFont();
```
Which should be fine if the `FontProxy` is `RefCounted`:
```
Ref font = this->fontProxy();
save();
font->metricsOfPrimaryFont();
```

Reply:
That seems to make sense. I think it was probably the right idea but wrong execution. The original PR had a lot of issues (which I commented on).